app/src/org/commcare/dalvik/preferences/CommCarePreferences.java

```diff
@@ -117,6 +129,63 @@ protected void onCreate(Bundle savedInstanceState) {
         this.getPreferenceScreen().addPreference(lp);
         updatePreferencesText();
         setTitle("CommCare" + " > " + "Application Preferences");
+
+        //Set an OnPreferenceClickListener for Print doc location
+        Preference pref = prefMgr.findPreference(PREF_MANAGER_PRINT_KEY);
+        pref.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
+            @Override
+            public boolean onPreferenceClick(Preference preference) {
+                if (preference.getKey().equals(PREF_MANAGER_PRINT_KEY)) {
+                    startFileBrowser();
+                    return true;
+                }
+                return false;
+            }
+        });
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+
+        if (requestCode == REQUEST_TEMPLATE) {
+            if (resultCode == RESULT_OK && data != null) {
+                Uri uri = data.getData();
+                String filePath = UriToFilePath.getPathFromUri(CommCareApplication._(), uri);
+                String extension = TemplatePrinterUtils.getExtension(filePath);
+                if (extension.equalsIgnoreCase("html")) {
+                    SharedPreferences.Editor editor = CommCareApplication._().getCurrentApp().
+                            getAppPreferences().edit();
+                    editor.putString(PRINT_DOC_LOCATION, filePath);
+                    editor.commit();
+                    Toast.makeText(this, Localization.get("template.success"), Toast.LENGTH_SHORT).show();
+                } else {
+                    showAlertDialog(Localization.get("template.not.set"), Localization.get("template.warning"));
+                }
+            } else {
+                //No file selected
+                Toast.makeText(this, Localization.get("template.not.set"), Toast.LENGTH_SHORT).show();
+            }
+        }
+
+    }
+
+    private void showAlertDialog(String title, String message) {
```

Comment:
Looks like this is also in TemplatePrinterActivity; might be worth putting it into a util class.